### PR TITLE
[java] Use HTTP interface for parametric test

### PR DESCRIPTION
**Pull request by cbeauchesne**
## Motivation

using HTTP instead of gRPC should remove most of the flakiness issues

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [x] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [x] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [x] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [x] A docker base image is modified?
    * [x] the relevant `build-XXX-image` label is present
* [x] A scenario is added (or removed)?
    * [x] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

**Review by cbeauchesne — COMMENTED** [review]
I cannot approve my own PR, so this is a soft approval from myself :) 

Could you get a review from someone from java team, to at least have a second eyes a java code?

Then all good to me. Thank's a lot for tackling this !

File: utils/build/docker/java/parametric/run.sh [hunk 1]
```diff
@@ -8,9 +8,16 @@ case $(echo "${DD_TRACE_DEBUG:-false}" | tr '[:upper:]' '[:lower:]') in
   *);;
 esac
 
+# TODO Need to disable all instrumentations related to Spring Boot
```
Comment (vandonr) [inline utils/build/docker/java/parametric/run.sh:11]: I think this line can be removed now ?
Maybe we can convert it into a comment explaining _why_ we disable Spring boot instrumentations. It's obvious in the context of this review, but might not be when looking solely at this file in the future.

Comment (PerfectSlayer) [reply utils/build/docker/java/parametric/run.sh:11]: Yes, that is done but you raised a good point. 
I will document the new flags bellow 👍

File: utils/build/docker/java/parametric/src/main/java/com/datadoghq/ApmTestClient.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@SpringBootApplication
+public class ApmTestClient {
+    private static final String SERVER_PORT_ARG = "APM_TEST_CLIENT_SERVER_PORT";
+    public static final Logger LOGGER = LoggerFactory.getLogger(ApmTestClient.class.getName());
```
Comment (mcculls) [inline utils/build/docker/java/parametric/src/main/java/com/datadoghq/ApmTestClient.java:13]:
```suggestion
    private static final Logger LOGGER = LoggerFactory.getLogger(ApmTestClient.class);
```

**Review by mcculls — APPROVED** [review]

**Comment by cbeauchesne** [comment]
@PerfectSlayer is it ok for you? if yes, I'll merge it with my admin powers.

**Comment by PerfectSlayer** [comment]
Please all-mighty @cbeauchesne 🙇

**Pull request merged**